### PR TITLE
feat(multi actions): name and description on action/generation configs

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

needed for https://github.com/dust-tt/dust/issues/4602

The upcoming [use-tools](https://dust.tt/w/78bda07b39/a/0e9889c787) dust app is quite similar to the existing "generate inputs" one, with the exception that it takes several specifications as input and lets the model choose between them. In order to have the model pick, we need a good description for each action and a unique name. 

We'll likely get those from the assistant builder. I added name/description fields to all actions + the generation. Optional currently

## Risk

Critical path + migration

## Deploy Plan

initdb (prodbox) then deploy

### CODE REVIEW

**Review by PopDaph — APPROVED** [review]
LGTM

**Review by spolu — APPROVED** [review]

File: types/src/front/assistant/templates.ts [hunk 1]
```diff
@@ -219,6 +223,8 @@ export function getAgentActionConfigurationType(
         type: "dust_app_run_configuration",
         appWorkspaceId: "template",
         appId: "template",
+        name: "run_dust_app",
+        description: "Run a Dust app.",
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply types/src/front/assistant/templates.ts:227]: No not used (yet ?)

File: front/lib/api/assistant/global_agents.ts [hunk 2]
```diff
@@ -887,6 +917,8 @@ async function _getDustGlobalAgent(
           workspaceId: prodCredentials.workspaceId,
           filter: { tags: null, parents: null },
         })),
+        name: "search_data_sources",
+        description: `Search the workspace's data sources.`,
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply front/lib/api/assistant/global_agents.ts]: fixd

File: front/lib/api/assistant/global_agents.ts [hunk 3]
```diff
@@ -648,6 +674,8 @@ async function _getManagedDataSourceAgent(
           workspaceId: prodCredentials.workspaceId,
           filter: { tags: null, parents: null },
         })),
+        name: "search_data_sources",
+        description: `Search the workspace's ${connectorProvider} data sources.`,
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply front/lib/api/assistant/global_agents.ts]: fixe

**Pull request merged**